ci-operator/step-registry/storage/create/csi-secrets-store/storage-create-csi-secrets-store-commands.sh

```diff
@@ -0,0 +1,121 @@
+kind: ClusterCSIDriver
+metadata:
+    name: ${CLUSTER_CSI_DRIVER_NAME}
+spec:
+  managementState: Managed
+EOF
+echo "Created ClusterCSIDriver ${CLUSTER_CSI_DRIVER_NAME}"
+
+# e2e-provider must be privileged to bind to a unix domain socket on the host.
+echo "Creating E2E Provider ServiceAccount"
+oc create serviceaccount -n ${E2E_PROVIDER_NAMESPACE} ${E2E_PROVIDER_SERVICE_ACCOUNT}
+oc adm policy add-scc-to-user privileged system:serviceaccount:${E2E_PROVIDER_NAMESPACE}:${E2E_PROVIDER_SERVICE_ACCOUNT}
+echo "Created E2E Provider ServiceAccount"
+
+echo "Creating E2E Provider DaemonSet"
+cat <<EOF >>${E2E_PROVIDER_DAEMONSET_LOCATION}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: ${E2E_PROVIDER_APP_LABEL}
+  name: csi-secrets-store-e2e-provider
+  namespace: ${E2E_PROVIDER_NAMESPACE}
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: ${E2E_PROVIDER_APP_LABEL}
+  template:
+    metadata:
+      labels:
+        app: ${E2E_PROVIDER_APP_LABEL}
+    spec:
+      serviceAccountName: ${E2E_PROVIDER_SERVICE_ACCOUNT}
+      containers:
+        - name: e2e-provider
+          image: ${SECRETS_STORE_E2E_PROVIDER_IMAGE}
+          imagePullPolicy: IfNotPresent
+          args:
+            - --endpoint=unix:///provider/e2e-provider.sock
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: "/provider"
+              name: providervol
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: type
+                operator: NotIn
+                values:
+                - virtual-kubelet
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/var/run/secrets-store-csi-providers"
+      nodeSelector:
+        kubernetes.io/os: linux
+EOF
+
+echo "Using E2E Provider DaemonSet file ${E2E_PROVIDER_DAEMONSET_LOCATION}"
+cat ${E2E_PROVIDER_DAEMONSET_LOCATION}
+
+oc create -f ${E2E_PROVIDER_DAEMONSET_LOCATION}
+echo "Created E2E Provider DaemonSet from file ${E2E_PROVIDER_DAEMONSET_LOCATION}"
+
+export E2E_PROVIDER_GET_ARGS="-n ${E2E_PROVIDER_NAMESPACE} --selector=${E2E_PROVIDER_SELECTOR}"
+echo "Waiting for E2E Provider pods to be created"
+loop_counter=0
+while true; do
+	oc get pods ${E2E_PROVIDER_GET_ARGS} && break
+	if [ ${loop_counter} -eq 60 ]; then
+		echo "Timed out while waiting for E2E Provider pods to be created"
+		exit 1
+	fi
+	loop_counter=${loop_counter}+1
+	sleep 5
+done
+
+echo "Waiting for E2E Provider pods to be Ready"
+if ! oc wait pod ${E2E_PROVIDER_GET_ARGS} --for=condition=Ready --timeout=300s; then
+	echo "Wait failed, E2E Provider pods did not reach Ready state"
+	exit 1
+fi
```

Comment:
you can wait for DaemonSet numberReady to be equal to nr. of nodes:

`oc -n openshift-cluster-csi-drivers wait --for=jsonpath='{.status.numberReady}'=$NODES daemonset/aws-ebs-csi-driver-node`

BTW, for any error case, you should dump the DaemonSet yaml for debugging.

Reply:
Cool trick, I pushed a new commit with this change.